_posts/2018-05-08-CocoaPods-1.6.0-beta.markdown

```diff
@@ -0,0 +1,84 @@
+
+<!-- more -->
+
+We are happy to announce that the first beta of 1.6.0 is out! For this release, we are revamping the build settings generation to be faster and easier to maintain. We've also improved integration of test specs for pods that have multple ones.
+
+### Revamped Build Settings
+
+The generation of build settings has been completely re-written. The primary goals for this re-write was to clean up the code but also allow this process to scale better on large projects. Having to apply a fix would often cause a cascading amount of performance regressions making it very hard to test but also apply the fix for everyone.
+
+From a user point of view, this should have zero effects when you compile your code. If there is anything that does not look right please file an [issue](https://github.com/CocoaPods/CocoaPods/issues/new) with us providing steps and a sample app showing the problem.
+
+#### LinkedIn Performance Case Study?
+
+TODO MAYBE INCLUDE THIS HERE
+
+### Better Test Spec Integration
+
+Test specs are becoming a fundamental piece of CocoaPods for pod authors. The ability to specify you test sources and have them execute through `lint` becomes an invaluable tool for making your `podspec` file source of truth for your pod. With 1.6.0, specifying multiple `test_spec` entries in your `podspec` will no longer merge them together under a single test bundle target.
+
+Let's use the following example:
+
+```ruby
+Pod::Spec.new do |s|
+  # ... rest of root spec entries go here
+  
+  # Unit Test Sources - Those do not require an app host to run. 
+  # They also require 'OCMock' dependency.
+  s.test_spec 'Tests' do |test_spec|
+	test_spec.source_files = 'Tests/**/*.{h,m}'
+	test_spec.dependency 'OCMock'
+  end
+
+  # SnapShot Tests Sources - Those *do* require an app host to run.
+  s.test_spec 'SnapshotTests' do |test_spec|
+   	test_spec.requires_app_host = true
+	test_spec.source_files = 'SnapshotTests/**/*.{h,m}'
+  end
+```
+
+Before 1.6.0, both the 'Tests' and 'SnapshotTests' test specs would _merge_ into a single test bundle target. This also has the unwanted effect of using an app host for the 'Tests' test spec without explicitly having declared.
+
+It looks like this:
+
+<center>
+{% breaking_image /assets/blog_img/CocoaPods-1.6.0-beta/test_specs_before.png %}
+</center>
+
+Furthermore, pod authors would find it unnecessary to split their test sources and categorize them in different ways, for example using a different set of dependencies or resources.
+
+And now it looks like this:
+
+<center>
+{% breaking_image /assets/blog_img/CocoaPods-1.6.0-beta/test_specs_after.png %}
+</center>
+
+Much better!
+
+We _strongly_ support the idea of having tests for your code and aim to give the tools to support you doing so. Having your `.podspec` file be the source of truth and coupled with recently released [cocoapods-generate](https://github.com/square/cocoapods-generate) plugin your developer experience can be much improved.
```

Comment:
This is great 👍 - good show at pushing the community towards higher standards

Perhaps "and aim to give the tools to support you doing so" could be more aspirational?

- "and aim to provide the right tools for doing this simply"